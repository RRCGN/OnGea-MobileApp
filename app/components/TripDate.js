import React from 'react'
import { View, Text, StyleSheet } from 'react-native'

const styles = StyleSheet.create({
  typeText: {
    fontSize: 12,
    color: 'rgba(0,0,0,0.54)'
  },
  dateText: {
    fontSize: 16,
    fontWeight: '500',
    color: 'rgba(0,0,0,0.87)'
  }
})

const TripDate = ({ typeText, dateText }) => {
  return (
    <View>
      <Text style={styles.typeText}>{typeText}</Text>
      <Text style={styles.dateText}>{dateText}</Text>
    </View>
  )
}

export default TripDate
