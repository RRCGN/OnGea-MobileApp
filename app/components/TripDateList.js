import React from 'react'
import { View, StyleSheet } from 'react-native'

const styles = StyleSheet.create({
  row: {
    flexDirection: 'row',
    marginLeft: -16,
    marginRight: -16
  },
  item: {
    paddingLeft: 16,
    paddingRight: 16,
    paddingTop: 8,
    paddingBottom: 8
  },
  itemSeparator: {
    borderLeftWidth: 1,
    borderLeftColor: '#E0E0E0'
  }
})

const TripDateList = ({ children }) => {
  return (
    <View style={styles.row}>
      {children.map((child, i) => {
        const styleList = [
          styles.item,
          i != 0 && styles.itemSeparator
        ]
        return <View key={i} style={StyleSheet.flatten(styleList)}>{child}</View>
      })}
    </View>
  )
}

export default TripDateList
