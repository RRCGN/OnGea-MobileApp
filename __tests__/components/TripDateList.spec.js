import { View } from 'react-native'
import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import TripDate from '../../app/components/TripDate'
import TripDateList from '../../app/components/TripDateList'

import renderer from 'react-test-renderer'

describe('<TripDateList />', () => {
  test('renders correctly', () => {
    const wrapper = shallow(
      <TripDateList>
        <TripDate type="vom" date="2017-04-10T02:00:00+02:00" />
        <TripDate type="vom" date="2017-04-10T02:00:00+02:00" />
      </TripDateList>
    )
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
