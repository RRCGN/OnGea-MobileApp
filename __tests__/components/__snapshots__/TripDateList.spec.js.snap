exports[`<TripDateList /> renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginLeft": -16,
      "marginRight": -16,
    }
  }>
  <View
    style={
      Object {
        "paddingLeft": 16,
        "paddingRight": 16,
      }
    }>
    <TripDate
      date="2017-04-10T02:00:00+02:00"
      type="vom" />
  </View>
  <View
    style={
      Object {
        "paddingLeft": 16,
        "paddingRight": 16,
      }
    }>
    <TripDate
      date="2017-04-10T02:00:00+02:00"
      type="vom" />
  </View>
</View>
`;
